web/hooks/push.go

```diff
@@ -68,48 +69,23 @@ func (wh GitHubWebHook) handlePush(payload *github.PushEvent) {
 		}
 		wh.logger.Debugf("Successfully cloned assignments repository to: %s", clonedAssignmentsRepo)
 
-	case repo.IsUserRepo():
-		wh.logger.Debugf("Processing push event for user repo %s", payload.GetRepo().GetName())
-		if !isDefaultBranch(payload) {
-			wh.logger.Debugf("Ignoring push event for non-default branch: %s", payload.GetRef())
-			return
-		}
-		wh.updateLastActivityDate(repo.UserID, course.ID)
+	case repo.IsStudentRepo():
+		wh.logger.Debugf("Processing push event for repo %s", payload.GetRepo().GetName())
 		assignments := wh.extractAssignments(payload, course)
 		for _, assignment := range assignments {
-			if !assignment.IsGroupLab {
-				// only run non-group assignments
-				wh.runAssignmentTests(sc, assignment, repo, course, payload)
-			} else {
-				wh.logger.Debugf("Ignoring push to user repo %s for group assignment: %s", payload.GetRepo().GetName(), assignment.GetName())
+			results := wh.runAssignmentTests(sc, assignment, repo, course, payload)
+			// Non-default branch indicates push to a group repo.
+			if !isDefaultBranch(payload) && !assignment.GradedManually() {
```

Comment:
If you move this code into `runAssignmentTests` you can remove the `GradedManually()` check (it is already done in `runAssignmentTests`) and we can remove the `results` return value from `runAssignmentTests`.

Reply:
The loop becomes nice and tidy:
```go
for _, assignment := range assignments {
	wh.runAssignmentTests(sc, assignment, repo, course, payload)
}
```

Reply:
The check at the end of the `runAssignmentTests` func becomes:
```go
	// Non-default branch indicates push to a group repo.
	if !isDefaultBranch(payload) {
		// Attempt to find the pull request for the branch, if it exists,
		// and then assign reviewers to it, if the branch task score is higher than the assignment score limit
		wh.handlePullRequestPush(payload, results, assignment, course, repo)
	}
```

Reply:
And you can remove all the returns.